rcl_lifecycle/src/com_interface.c

```diff
@@ -0,0 +1,203 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdio.h>
+#include <string.h>
+
+#include "rosidl_generator_c/message_type_support.h"
+#include "rosidl_generator_c/string_functions.h"
+
+#include "lifecycle_msgs/msg/transition_event.h"
+#include "lifecycle_msgs/srv/get_state.h"
+#include "lifecycle_msgs/srv/change_state.h"
+
+#include "rcl/error_handling.h"
+
+#include "rcl_lifecycle/data_types.h"
+
+#include "com_interface.hxx"
+
+static lifecycle_msgs__msg__TransitionEvent msg;
+
+bool concatenate(const char ** prefix, const char ** suffix, char ** result)
```

Comment:
`strcat` and `strcat_s` requires branching between platforms. I haven't looked into sscanf though.